server.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express')
+const path = require('path')
+const app = express()
+const port = process.env.PORT || 5001
+
+// Dynamic configuration for openshift API calls
+app.get('/config.js', (req, res) => {
+  // The x-forwarded-access-token header will be forwarded
+  // by the oauth-proxy to this server.
+  // The below makes this token available to the client app
+  // for use in API calls to the configured OpenShift cluster.
+  res.send(`window.OPENSHIFT_CONFIG = {
+    openshiftAPIUrl: "${process.env.OPENSHIFT_API_URL}",
+    accessToken: "${req.headers['x-forwarded-access-token']}"
+  }`)
+})
+
+if (process.env.NODE_ENV === 'production') {
+  // Serve any static files
+  app.use(express.static(path.join(__dirname, 'build')))
+  // Handle React routing, return all requests to React app
+  app.get('*', function (req, res) {
```

Comment:
this would enable us to use `BrowserRouter` as opposed to `HashRouter` from a react-router perspective if desired. 👍

Reply:
For the oauth redirect flow, BrowserRouter would be preferrable.